### PR TITLE
[JDF-288] - Add a quickstart that shows authentication and authorization against LDAP.

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

**Comment by sgilda** [comment]
When I run 'mvn -Dtest=LDAPServer test', I see a lot of errors like the following:

```
    Cannot normalize the value :ERR_04226 I do not know how to handle NameAndOptionalUID normalization with objects of class: <some class name>
```

Is that to be expected? If so, it would be nice to mention that they can be ignored.

**Comment by sgilda** [comment]
I deployed an am able to access the application, however none of the logins work. I get "Authentication was unsuccessful. Please check your username and password before trying again." 

I don't see anything in the JBoss server log.

I see this in the command line running the embedded LDAP server: 

6462258 [pool-4-thread-4] ERROR org.apache.directory.server.ldap.handlers.LdapRequestHandler - ERR_732 Cannot process a Request while binding

**Comment by sgilda** [comment]
It looks like you must be in a directory with a POM file to execute the `mvn -Dtest=LDAPServer test` command, so up above, change step 2 to:

2, Navigate to the root of this quickstart directory and execute the following command:

**Comment by sgilda** [comment]
I tried restarted my LDAP and JBoss server and tried it again. It worked this time. I think the first time the browser filled in my admin password, which was different from the example, so it failed, and somehow I wasn't able to get in with the correct one.

Just curious why the package name for the source files end with the jsf package? ;-)

This one is ready for code review.

**Comment by pedroigor** [comment]
Removed the jsf package.
Updated README.md.
Added more comments.

**Comment by sgilda** [comment]
@pedroigor : 
1. In the README.md file, the quickstart name still shows this:

```
    picketlink-authorization-ldap: PicketLink IDM Authorization using LDAP
It should be:

    picketlink-authorization-idm-ldap: PicketLink IDM Authorization using LDAP
```
1. Under 'What is it?', could you add:

You can test this quickstart using the provided embedded LDAP server or you can configure this quickstart to connect to your own LDAP server. Instructions for both methods are provided below.
1. Could you move the entire section 'Configure LDAP' down after 'Configure Maven' or 'Start JBoss Enterprise  Application Platform 6 or JBoss AS 7 with the Web Profile', but before 'Build and Deploy the Quickstart'?
2. Are the errors expected when you start the embedded LDAP server? If so, we should tell them not to be concerned. 

I will make more comments in the individual files. :-)

File: picketlink-authorization-idm-ldap/README.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+picketlink-authorization-ldap: PicketLink IDM Authorization using LDAP
```
Comment (sgilda) [inline picketlink-authorization-idm-ldap/README.md]: This needs to be: picketlink-authorization-idm-ldap: PicketLink IDM Authorization using LDAP

File: picketlink-authorization-idm-ldap/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+What is it?
+-----------
+
+You'll learn from this example how to authenticate and authorize users(role-based) against a LDAP server,
+using a LDAP-based Identity Store.
+
+The latest PicketLink documentation is available [here](http://docs.jboss.org/picketlink/2/latest/).
+
+Configure LDAP
```
Comment (sgilda) [inline picketlink-authorization-idm-ldap/README.md]: As I said in the main comment area, this should be moved lower, before "Build and Deploy the Quickstart"

File: picketlink-authorization-idm-ldap/README.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+Target Product: EAP
+Source: <https://github.com/jboss-jdf/jboss-as-quickstart/>
+
+
+What is it?
+-----------
+
+You'll learn from this example how to authenticate and authorize users(role-based) against a LDAP server,
+using a LDAP-based Identity Store.
+
```
Comment (sgilda) [inline picketlink-authorization-idm-ldap/README.md:16]: Add:

You can test this quickstart using the provided embedded LDAP server or you can configure this quickstart to connect to your own LDAP server. Instructions for both methods are provided below.

File: picketlink-authorization-idm-ldap/README.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+### Configure the Quickstart to Use Your LDAP Server
+
+To run this quickstart using your own LDAP Server, modify the String constant values in the  `org.jboss.as.quickstarts.picketlink.idm.ldap.IDMConfiguration` class to the connection values for your LDAP server.
+
+### Configure the Quickstart to Use the Embedded LDAP Server
+
+To run this quickstart using an embedded LDAP server:
+
+1. Open a new command line.
```
Comment (sgilda) [inline picketlink-authorization-idm-ldap/README.md]: You need to have a pom file for the Maven command to run. Change this line to:
1. Open a new command line an navigate to the root of this quickstart directory.

**Comment by sgilda** [comment]
In the README:
The application will be running at the following URL: http://localhost:8080/jboss-as-picketlink-authorization-ldap. 

This should be:
The application will be running at the following URL: http://localhost:8080/jboss-as-picketlink-authorization-idm-ldap. 

In the pom.xml file, this:
<artifactId>jboss-as-picketlink-authorization-ldap</artifactId>

Should be:
<artifactId>jboss-as-picketlink-authorization-idm-ldap</artifactId>

**Comment by sgilda** [comment]
Once I enter an invalid userid/password combination, it won't work when I enter a valid one unless I restart the server. Is that the expected behavior?

**Comment by pedroigor** [comment]
This is not an expected behavior. I did some tests and the authentication is fine even when providing an invalid user before.

Can you describe the your steps, please ?

**Comment by pedroigor** [comment]
As discussed before, we're doing some changes to PL. If you want to hold this PR until we release a new version, for me is fine.

The reason is that those changes will impact the quickstarts. So we probably need to review them anyway. Both merged and unmerged quickstarts.

Going to let you know once we get them updated.

Thanks.

**Comment by sgilda** [comment]
Closing this pull as it is superceded by a pull to create a submodule for the PicketLink quickstarts: https://github.com/jboss-jdf/jboss-as-quickstart/pull/584.

**Pull request closed without merging**